php-checks/src/test/resources/checks/regex/InvalidRegexCheck.php

```diff
@@ -30,5 +30,7 @@ function compliant($input)
   {
     preg_match('/$[a-z]^/', $input);
     preg_replace('/[^A-Za-z0-9\-\][\^!@#$%&*)(+=}{]/', '', $input);
+
+    preg_match("/((?:[\\0\\r\\n]|\u{202A}|\u{202B}|\u{202D}|\u{202E}|\u{2066}|\u{2067}|\u{2068}|\u{202C}|\u{2069})++)(.)/", $input);
```

Comment:
You could reduce the test case to a minimum to show on which minimal reproducer the issue was raised before the fix. And maybe show a similar test case where we still raise for an octal value.